[feature] add help window feature when ? is pressed

theghostmac opened the pull request.

radlinskii commented:
Hi, great job at adding the help window!

I have one suggestion, I think it should only be accessible Normal mode, so either during Pause, or before running the test.
Maybe also in the test results screen, but that can be a separate issue.

Guess the description wasn't clear enough, sorry about that.

theghostmac commented:
Hey @radlinskii, I just addressed that in the last 2 commits.

Nice project btw.

theghostmac commented:
Just did `cargo fmt` That should pass the `build` workflow.

radlinskii commented:
the `build` job run is still failing,  let me know if you plan to fix it yourself or not

radlinskii commented:
ref #70

theghostmac commented:
I'll fix it. I've seen the issue.

theghostmac commented:
Hi @radlinskii I've addressed the issue.

1. Changed the render method to take impl FrameWrapperInterface instead of Frame<B>
2. Removed the generic <B: Backend> since we're no longer using the raw Frame
3. Changed direct frame methods to use the FrameWrapperInterface methods
4. Simplified the imports since we don't need Backend anymore

theghostmac commented:
Hey @radlinskii , 
I'm thinking of updating the `cli.rs` to not ignore the tests as the pass now.

radlinskii commented:
did you even try to run the app with your changes?

radlinskii commented on src/help_window.rs:
```diff
@@ -48,16 +51,15 @@ impl HelpWindow {
         for (i, &text) in help_text.iter().enumerate() {
             let paragraph = Paragraph::new(text).style(Style::default().fg(Color::White));
-            frame.render_widget(paragraph, chunks[i]);
+            if i < chunks.len() {
+                frame.render_widget(paragraph, chunks[i]);
+            }
```
maybe consider simply creating the right amount of chunks, the same as the number of elements in help_text vector?

theghostmac replied:
done.

radlinskii commented:
when help window is rendered then 's' keypress should not start test, or if it did, than the help window should hide.

theghostmac commented:
Okay, the most user-friendly one is the econd option -- to automatically hide help when 's' is pressed.

The user can now:
- Toggle help with "?"
- Press 's' to dismiss help and start the test in one action
- Still use 'q' to quit whether help is shown or not

Any other changes required to improve the user experience?
